src/test/Fuzzers.sol

```diff
@@ -10,35 +10,95 @@ import {BalanceDelta} from "../types/BalanceDelta.sol";
 import {TickMath} from "../libraries/TickMath.sol";
 import {Pool} from "../libraries/Pool.sol";
 import {PoolModifyLiquidityTest} from "./PoolModifyLiquidityTest.sol";
+import {LiquidityAmounts} from "../../test/utils/LiquidityAmounts.sol";
+import {SafeCast} from "../../src/libraries/SafeCast.sol";
 
 contract Fuzzers is StdUtils {
+    using SafeCast for uint256;
+
     Vm internal constant _vm = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
 
-    function boundLiquidityDelta(PoolKey memory key, int256 liquidityDelta) internal pure returns (int256) {
-        return bound(
-            liquidityDelta, 0.0000001e18, int256(uint256(Pool.tickSpacingToMaxLiquidityPerTick(key.tickSpacing)) / 2)
+    function boundLiquidityDelta(PoolKey memory key, int256 liquidityDeltaUnbounded, int256 liquidityMaxByAmount)
+        internal
+        pure
+        returns (int256)
+    {
+        int256 liquidityMaxPerTick = int256(uint256(Pool.tickSpacingToMaxLiquidityPerTick(key.tickSpacing)));
+
+        // Finally bound the seeded liquidity by either the max per tick, or by the amount allowed in the position range.
+        int256 liquidityMax = liquidityMaxByAmount > liquidityMaxPerTick ? liquidityMaxPerTick : liquidityMaxByAmount;
+        return bound(liquidityDeltaUnbounded, 0.0000001e18, liquidityMax);
+    }
+
+    function boundLiquidityDeltaStrict(PoolKey memory key, int256 liquidityDeltaUnbounded, int256 liquidityMaxByAmount)
```

Comment:
can you add some comments explaining when one should use `boundLiquidityDeltaStrict` vs `boundLiquidityDelta`

Reply:
like this one just seems to divide by 2..? i dont understand what makes it "strict"

Reply:
This is the old version (ie how we were doing it on all of the StateLibrary tests.) 

We need a "strict" version for bounding when you want to add liquidity at potentially overlapping ticks. The liquidity boundaries that I added are for vanilla one off positions (ie it bounds it at the max for one tick) which will lead to reverts if there is already liquidity at that tick.

The 2 is carried over from before - it's not really exact and maybe could be better for multiple positions but it works for now

Reply:
I see I see. I feel like the function names could be changed to reflect this more clearly. Maybe just `boundLiquidityDeltaByMax` and `boundLiquidityDeltaForMultiplePositions` or `boundLiquidityDeltaByHalfMax` or something.

Reply:
sure can do that!